### ISSUE TITLE
Prisma CLI fails to connect to postgres in node:lts docker image

### ISSUE THREAD

**Issue by pabloogc** [prisma/prisma#13060]
### Bug description

Any command involving the CLI in the container with base image `node:lts` fail with the error: 

```Error: P1001: Can't reach database server at `postgres`:`5432```

The client itself inside the dockerized application connects normally (```client.$connect()``` works). The CLI commands from the host machine also work normally.

Upgrading the node version to 18 seems to fix, but it's not possible to do so in my current company environment.

### How to reproduce

Repo to reproduce: https://github.com/pabloogc/prisma-bug

Run:

1. `docker-compose build`
2. `docker-compose run prisma` (might take a bit to start until postgres is ready)
3. Verify postgres is ready from the container: `pg_isready -h postgres`
```bash
root@3694650b3fc1:/app# pg_isready -h postgres
postgres:5432 - accepting connections
```
4. Verify `DATABASE_URL` is properly configured
```bash
env
DATABASE_URL=postgres://postgres:qwerty@postgres:5432/main?schema=public
```
5. Run any CLI command in the container (`npm run prisma -- db push`), it almost always will fail:
```bash
root@3694650b3fc1:/app# npm run prisma -- db push

> prisma
> prisma "db" "push"

Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "main", schema "public" at "postgres:5432"

Error: P1001: Can't reach database server at `postgres`:`5432`

Please make sure your database server is running at `postgres`:`5432`.
```

7. Verify again that indeed the server is reachable connecting directly with psql:

```bash
root@3694650b3fc1:/app# psql postgres://postgres:qwerty@postgres:5432/main
psql (11.14 (Debian 11.14-0+deb10u1), server 14.2)
WARNING: psql major version 11, server major version 14.
         Some psql features might not work.
Type "help" for help.

main=# select version();
                                                      version                                                       
--------------------------------------------------------------------------------------------------------------------
 PostgreSQL 14.2 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 10.3.1_git20211027) 10.3.1 20211027, 64-bit
(1 row)

main=# 
```

### Expected behavior

The CLI works in the container.

### Prisma information

You can find it in the repo: https://github.com/pabloogc/prisma-bug

### Environment & setup

- OS: Debian GNU/Linux 10 (buster)
- Database: Postgres
- Node.js version: v16.14.2

### Prisma Version

```
prisma                  : 3.13.0
@prisma/client          : 3.13.0
Current platform        : linux-arm64-openssl-1.1.x
Query Engine (Node-API) : libquery-engine efdf9b1183dddfd4258cd181a72125755215ab7b (at ../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/libquery_engine-linux-arm64-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli efdf9b1183dddfd4258cd181a72125755215ab7b (at ../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/migration-engine-linux-arm64-openssl-1.1.x)
Introspection Engine    : introspection-core efdf9b1183dddfd4258cd181a72125755215ab7b (at ../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/introspection-engine-linux-arm64-openssl-1.1.x)
Format Binary           : prisma-fmt efdf9b1183dddfd4258cd181a72125755215ab7b (at ../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/prisma-fmt-linux-arm64-openssl-1.1.x)
Default Engines Hash    : efdf9b1183dddfd4258cd181a72125755215ab7b
Studio                  : 0.459.0
```

**Comment by oxyli6z** [comment]
My workaround on this is having a `DATABASE_URL` and a `PRISMA_URL` on the `.env` file, one referencing the name of the database container in this case `postgres`, and the other referencing `localhost`, like that.

`DATABASE_URL=postgres://postgres:postgres@postgres:5432/main`
and
`PRISMA_URL=postgres://postgres:postgres@localhost:5432/main`

That happens because the server container doesn't communicate with the database container using `localhost` inside the Docker network, only using the name of the container like `postgres`.
But since you're migrating from your CLI and not from inside the Docker network, you can use `localhost`.

Basically, you use `PRISMA_URL` for the `schema.prisma` file and `DATABASE_URL` for anything else.

Let me know if that helps.

**Comment by oxyli6z** [comment]
My workaround on this is having a DATABASE_URL and a PRISMA_URL on the .env file, one referencing the name of the database container in this case postgres, and the other referencing localhost, like that.

DATABASE_URL=postgres://postgres:postgres@postgres:5432/main
and
PRISMA_URL=postgres://postgres:postgres@localhost:5432/main

That happens because the server container doesn't communicate with the database container using localhost inside the Docker network, only using the name of the container like postgres.
But since you're migrating from your CLI and not from inside the Docker network, you can use localhost.

Basically, you use PRISMA_URL for the schema.prisma file and DATABASE_URL for anything else.

Let me know if that helps.

**Comment by oxyli6z** [comment]
My workaround on this is having a `DATABASE_URL` and a `PRISMA_URL` on the `.env` file, one referencing the name of the database container in this case `postgres`, and the other referencing `localhost`, like that.

`DATABASE_URL=postgres://postgres:postgres@postgres:5432/main`
and
`PRISMA_URL=postgres://postgres:postgres@localhost:5432/main`

That happens because the server container doesn't communicate with the database container using `localhost` inside the Docker network, only using the name of the container like `postgres`.
But since you're migrating from your CLI and not from inside the Docker network, you can use `localhost`.

Basically, you use `PRISMA_URL` for the `schema.prisma` file and `DATABASE_URL` for anything else.

Let me know if that helps.

**Comment by Nickersoft** [comment]
Would just like to hop in here and +1 this – seeing this behavior as well :/

**Comment by Seth-Carter** [comment]
I'm having the same issue. Using a Node 14 or Node 18 image will allow prisma to connect again.

**Comment by danarbaugh** [comment]
I am also seeing this behavior between node:16 and postgres:14 containers currently. Changing to node:18 on the connecting container with zero other changes results in successful CLI connections.

Prisma version:
```
prisma                  : 4.5.0
@prisma/client          : 4.5.0
Current platform        : linux-arm64-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/libquery_engine-linux-arm64-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/migration-engine-linux-arm64-openssl-1.1.x)
Introspection Engine    : introspection-core 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/introspection-engine-linux-arm64-openssl-1.1.x)
Format Binary           : prisma-fmt 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/prisma-fmt-linux-arm64-openssl-1.1.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.5.0-43.0362da9eebca54d94c8ef5edd3b2e90af99ba452
Default Engines Hash    : 0362da9eebca54d94c8ef5edd3b2e90af99ba452
Studio                  : 0.476.0
```

Postgres version
```
                                                              version                                                              
-----------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 14.5 (Debian 14.5-1.pgdg110+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
(1 row)
```

**Comment by jkomyno** [comment]
Closing this as a duplicate of https://github.com/prisma/prisma/issues/16182, which has a working solution.

**TL/DR**: `node:14`, `node:16`, `node:16-slim` all use Linux Debian 10.x (Buster) under the hood, but don't have any `openssl` version installed by default, which Prisma needs in order to work correctly.
Either upgrade to `node:18-slim` (which is built on Linux Debian 11.x, Bullseye), or install `openssl` manually:

```Dockerfile
RUN apt-get update -y \
  && apt-get install -y openssl
```

Please see [this comment](https://github.com/prisma/prisma/issues/16182#issuecomment-1371283457) for more context about the solution and why this issue happened in the first place.
